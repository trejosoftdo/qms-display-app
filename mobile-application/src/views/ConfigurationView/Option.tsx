import * as React from 'react';
import { List } from 'react-native-paper';

/**
 * OptionProps defines the props for the Option Component.
 *
 * @property title - option title
 * @property icon - option icon
 * @property onPress - option on press handler
 */
interface OptionProps {
  title: string;
  icon: string;
  onPress: () => void;
}

/**
 * A component a configuration option
 *
 * @param {OptionProps} props - The props for the Option component.
 */
const Option: React.FC<OptionProps> = (props: OptionProps) => (
  <List.Item
    title={props.title}
    left={() => <List.Icon icon={props.icon} />}
    right={() => <List.Icon icon="chevron-right" />}
    onPress={props.onPress}
  />
);

export default Option;