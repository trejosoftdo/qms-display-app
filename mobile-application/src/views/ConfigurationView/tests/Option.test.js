import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import Option from '../Option';

describe('Option component', () => {
  const mockProps = {
    title: 'mock-title',
    icon: 'lock',
    onPress: jest.fn(),
  };

  afterEach(() => {
    jest.clearAllMocks();
  })

  it('renders the component as expected', () => {
    expect(renderShallow(<Option {...mockProps} />)).toMatchSnapshot();
    expect(mockProps.onPress).toHaveBeenCalledTimes(0);
  });

  it('calls the onPress handler when being clicked', () => {
    const { getByText } = render(<Option {...mockProps} />);
    const element = getByText(mockProps.title);
    fireEvent.press(element);
    expect(mockProps.onPress).toHaveBeenCalled();
  });
});
