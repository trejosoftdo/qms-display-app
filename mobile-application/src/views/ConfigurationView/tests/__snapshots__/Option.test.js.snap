// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option component renders the component as expected 1`] = `
<List.Item
  left={[Function]}
  onPress={[MockFunction]}
  right={[Function]}
  title="mock-title"
/>
`;
